test/sequential/test-fs-readdir-recursive.js

```diff
@@ -131,6 +131,7 @@ function getDirentPath(dirent) {
 }
 
 function assertDirents(dirents) {
+  assert.strictEqual(dirents.length, expected.length);
   dirents.sort((a, b) => (getDirentPath(a) < getDirentPath(b) ? -1 : 1));
   for (const [i, dirent] of dirents.entries()) {
     assert(dirent instanceof fs.Dirent);
```

Comment:
Can you also check if any of these dirents does not have any undefined value? It seems like one of the comments included the mention of `undefined` name value for a Dirent entry.